File: DSA/akashyadav_590014126/Day_37_Question_2.py
def evaluate_postfix(expr):
    stack=[]

    for ch in expr:
        if ch.isdigit():
            stack.append(int(ch))
        else:
            b=stack.pop()
            a=stack.pop()
        
        if ch=='+':
            stack.append(a+b)

        if ch=='-':
            stack.append(a-b)

        if ch=='*':
            stack.append(a*b)

        if ch=='/':
            stack.append(a/b)
    
    return stack[-1]

#example
print(evaluate_postfix("78-"))

File: DSA/RATNA_KHANNA_590013141/DAY_37_QUESTION_2.c
#include <stdio.h>
#include <ctype.h>  
#include <stdlib.h>
#define MAX 100
int stack[MAX];
int top = -1;
void push(int val) {
    if (top >= MAX - 1) {
        printf("Stack Overflow\n");
        exit(1);
    }
    stack[++top] = val;
}

int pop() {
    if (top < 0) {
        printf("Stack Underflow\n");
        exit(1);
    }
    return stack[top--];
}

int evaluatePostfix(const char* expr) {
    int i = 0;
    char ch;

    while ((ch = expr[i]) != '\0') {
        if (ch == ' ') {
            i++;
            continue;
        }

        if (isdigit(ch)) {
            push(ch - '0');
        } else {
            int b = pop();
            int a = pop();
            int result;

            switch (ch) {
                case '+': result = a + b; break;
                case '-': result = a - b; break;
                case '*': result = a * b; break;
                case '/': result = a / b; break;
                default:
                    printf("Unsupported operator: %c\n", ch);
                    exit(1);
            }
            push(result);
        }
        i++;
    }

    return pop();  
}

int main() {
    char expr[MAX];

    printf("Enter a postfix expression ");
    scanf("%s", expr);  

    int result = evaluatePostfix(expr);
    printf("Result: %d\n", result);

    return 0;
}
